common/djangoapps/dark_lang/views.py

```diff
@@ -0,0 +1,153 @@
+"""
+Views file for the Darklang Django App
+"""
+from openedx.core.lib.api.view_utils import view_auth_classes
+from django.views.generic.base import View
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+
+from edxmako.shortcuts import render_to_response
+
+from openedx.core.djangoapps.user_api.preferences.api import (
+    delete_user_preference, get_user_preference, set_user_preference
+)
+from lang_pref import LANGUAGE_KEY
+
+from django.utils.translation import LANGUAGE_SESSION_KEY
+from django.utils.translation import ugettext as _
+
+from dark_lang import DARK_LANGUAGE_KEY
+from dark_lang.models import DarkLangConfig
+
+LANGUAGE_INPUT_FIELD = 'preview_lang'
+
+
+@view_auth_classes()
+class DarkLangView(View):
+    """
+    View used when a user is attempting to change the preview language using Darklang.
+
+    Expected Behavior:
+    GET - returns a form for setting/resetting the user's dark language
+    POST - updates or clears the setting to the given dark language
+
+    Note: This class and View are meant to replace the ability to set the DarkLang in the middleware, which was
+    determined to be a security risk. (TNL-4742)
```

Comment:
Ha! I was tempted to comment on that as well, but decide to see if @nasthagiri brought it up. I'm used to leaving such artifacts in the code at other companies, but edX doesn't seem to do it (perhaps because the code is open source?).